model/plugin_kvset_options.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type PluginKVSetOptions struct {
+	EncodeJSON      bool
+	Atomic          bool
+	OldValue        interface{}
+	ExpireInSeconds int64
+}
+
+func (opt *PluginKVSetOptions) IsValid() *AppError {
+	if !opt.Atomic && opt.OldValue != nil {
```

Comment:
rather than checking the validity of a parameter combination like this should we use the `Functional Options` pattern to solve this instead? thoughts? see my earlier comment on the API.